Change the logic of detection

para0x0dise opened the pull request:
The signature is only triggered by the `NtReadFile` API, which requires the file to exist; however, if the browser is not installed on the guest VM, the signature won't be triggered. Most info stealers first check for the file's existence using the `PathFileExists` API, which internally calls the `NtQueryAttributesFile` API.

Also, I have added a counter to make the signature fired up if it hits a defined threshold, not in general to avoid false-positives.

doomedraven commented:
looks good to me, but lets wait for @kevoreilly review too

kevoreilly commented:
I may be being a n00b here but are these 'double datas' intended?

```
            # Google Chrome
            re.compile(".*\\\\Application\\ Data Data\\\\Google\\\\Chrome\\\\.*", re.I),
            re.compile(".*\\\\Local\\\\Google\\\\Chrome\\\\User\\ Data\\\\Default\\\\.*", re.I),
            re.compile(".*\\\\AppData\\\\Local\\\\Google\\\\Chrome\\\\User\\ Data\\\\Default\\\\.*", re.I),

            # Chromium-based Browsers
            re.compile(".*\\\\Application\\ Data Data\\\\Chromium\\\\.*", re.I),
            re.compile(".*\\\\AppData\\\\Local\\\\Chromium\\\\.*", re.I),
            re.compile(".*\\\\Application\\ Data Data\\\\ChromePlus\\\\.*", re.I),
            re.compile(".*\\\\AppData\\\\Local\\\\MapleStudio\\\\ChromePlus\\\\.*", re.I),
            re.compile(".*\\\\Application\\ Data Data\\\\Nichrome\\\\.*", re.I),
            re.compile(".*\\\\Application\\ Data Data\\\\Bromium\\\\.*", re.I),
            re.compile(".*\\\\Application\\ Data Data\\\\RockMelt\\\\.*", re.I),
            re.compile(".*\\\\Application\\ Data Data\\\\Flock\\\\.*", re.I),
            re.compile(".*\\\\AppData\\\\Local\\\\Flock\\\\.*", re.I),
            re.compile(".*\\\\Application\\ Data Data\\\\Comodo\\\\Dragon\\\\.*", re.I),
            re.compile(".*\\\\AppData\\\\Local\\\\Comodo\\\\Dragon\\\\.*", re.I),
            re.compile(".*\\\\BraveSoftware\\\\Brave-Browser\\\\User\\ Data\\\\Default\\\\.*", re.I),

            # Opera
            re.compile(".*\\\\Application\\ Data Data\\\\Opera\\\\.*", re.I),
```

doomedraven commented:
Yes, if no r prefix is used

doomedraven commented:
why Data is duplicated?

para0x0dise commented:
> why Data is duplicated? i mean `Application\\ Data Data`

Hello, Sorry for the delay, You are right, it's my mistake.

doomedraven commented:
ok cool thanks, im doing some fixes to general sigs before im leaving to PTO as i just got too many deprecation errors on my server :D

doomedraven commented:
ah and i have reitroduced some logic back, so we can properly cover it

```
if call["api"] == "NtReadFile":
            filename = self.get_argument(call, "HandleName")
        elif call["api"] == "NtQueryAttributesFile":
            filename = self.get_argument(call, "FileName")
        else:
            filename = self.get_argument(call, "ExistingFileName")
```

doomedraven commented:
check it now plz, if it looks fine for you then im merging it

para0x0dise commented:
> check it now plz, if it looks fine for you then im merging it

I have modified some regex, I don't remember why I use `\\User \\Data` instead of `\\User Data` :), and I have added new paths to include Appdata

doomedraven commented:
thanks, looks fine to me, sorry for delay
